src/erldns_zone_cache.erl

```diff
@@ -223,9 +263,59 @@ put_zone(Name, Zone) ->
 put_zone_records(Name, RecordsByName) ->
   put_zone_records_entry(Name, maps:next(maps:iterator(RecordsByName))).
 
+%% @doc Put zone RRSet
+%-spec put_zone_rrset({Name, Sha, Records, Keys}, Type, Counter | {Name, Sha, Records}, Type, Counter) -> ok | {error, Reason :: term()}
+%  when Name :: binary(), Sha :: binary(), Records :: [dns:rr()], Keys :: [erldns:keyset()], RRFqdn :: binary(), Type :: binary(), Counter :: integer().
+put_zone_rrset({ZoneName, Sha, Records}, RRFqdn, RRSetType, Counter) ->
+  put_zone_rrset({ZoneName, Sha, Records, []}, RRFqdn, RRSetType, Counter);
+put_zone_rrset({ZoneName, _Sha, Records, Keys}, RRFqdn, RRSetType, Counter) ->
+  % Check counter
+  CurrentCounter = get_sync_counter(),
+  lager:debug("Current Counter: ~p", [CurrentCounter]),
+  case CurrentCounter < Counter of 
+	  false -> 
+		  lager:debug("Not processing RRSet (~p) for Zone (~p): counter (~p) provided is lower than system", [RRFqdn, ZoneName, Counter]);
+	  true -> 
+		  lager:debug("Processing RRSet (~p) for Zone (~p): counter (~p) provided is higher than system", [RRFqdn, ZoneName, Counter]),
+		  % TODO: add custom types lookup
+		  Type = erldns_records:name_type(RRSetType),
+		  case find_zone_in_cache(erldns:normalize_name(ZoneName)) of
+			{ok, Zone} -> Zone,
+			  lager:debug("Putting RRSet (~p) with Type: ~p for Zone (~p): ~p", [RRFqdn, Type, ZoneName, Records]),
+			  lager:debug("Available Keys: ~p", [Keys]),
+			  % if keys are missing from RRset payload, use existing zone keysets
+			  KeySets = case length(Keys) of
```

Comment:
For simplicity always use the keys from the zone for now.

Reply:
commit: https://github.com/dnsimple/erldns/commit/fff79c3ec3cabb1b1783d528533ee0e7f704942d 

covers this (use existing Zone keysets only)